jira-macro/jira-macro-common/src/main/java/org/xwiki/contrib/jira/macro/internal/JIRABadRequestException.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.contrib.jira.macro.internal;
+
+import java.util.List;
+
+/**
+ * Exception thrown when there is an error when accessing the remote JIRA server.
+ *
+ * @version $Id$
+ * @since 11.0.1
+ */
+public class JIRABadRequestException extends JIRAConnectionException
```

Comment:
Shouldn't it exposed as an API instead of remain internal?

Reply:
We can expose as an API but in this case I suppose we might need also to expose `JIRAConnectionException` as an API ?

Reply:
Yes in such case you'd need to...